Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -303,64 +320,89 @@ public object Deserialize(IIonReader reader, Type type, IonType ionType)
 
             if (ionType == IonType.Bool)
             {
-                return new IonBooleanSerializer().Deserialize(reader);
+                var serializer = this.GetCustomPrimitiveSerializer<bool>() ?? new IonBooleanSerializer();
+                return serializer.Deserialize(reader);
             }
 
             if (ionType == IonType.Int)
             {
                 if (reader.GetTypeAnnotations().Any(s => s.Equals(IonLongSerializer.ANNOTATION)))
                 {
-                    return new IonLongSerializer().Deserialize(reader);
+                    var serializer = this.GetCustomPrimitiveSerializer<long>() ?? new IonLongSerializer();
+                    return serializer.Deserialize(reader);
+                }
+                else
+                {
+                    var serializer = this.GetCustomPrimitiveSerializer<int>() ?? new IonIntSerializer();
+                    return serializer.Deserialize(reader);
                 }
-                return new IonIntSerializer().Deserialize(reader);
             }
 
             if (ionType == IonType.Float)
             {
                 if (reader.GetTypeAnnotations().Any(s => s.Equals(IonFloatSerializer.ANNOTATION)))

```

Comment:
If this is a float, shouldn't it be treated as a float by default, and only be a double if it has the Double annotation? Similar to the IonType.Int path above.

Reply:
There is no `IonType.Double`.